src/dof.cpp

```diff
@@ -96,7 +96,7 @@ void DOF::updateMinMax()
   DBGP("Joint 0 min "  << minq << " max " << maxq);
   if (maxq < minq) std::swap(maxq, minq);  
   DBGP("maxq " << maxq << " minq " << minq);
-  std::list<Joint *>::iterator j;
+  std::vector<Joint *>::iterator j;
```

Comment:
This change from list to vector has required changes in a bunch of places; what was the initial reason for it?